### PR TITLE
Do not error if use_2to3 is set to a false value

### PR DESCRIPTION
**Pull request by plumdog**
## Summary of changes

I think this slight change fixes an issue where setting `use_2to3=False` would fail fast when really it ought to be ignored.

I think this results in behaviour closer to what I had in mind with #2769.

### Pull Request Checklist
- [ ] Changes have tests - no because I can't see where one would go
- [x] News fragment added in `changelog.d/`.

### CODE REVIEW

**Comment by haoyuanli** [comment]
should 58.0.2 be yanked? I assume this will go into a 58.0.3

File: setuptools/dist.py [hunk 1]
```diff
@@ -289,8 +289,11 @@ def assert_bool(dist, attr, value):
         raise DistutilsSetupError(tmpl.format(attr=attr, value=value))
 
 
-def invalid(dist, attr, value):
-    raise DistutilsSetupError(f"{attr} is invalid.")
+def invalid_ignored_if_false(dist, attr, value):
+    if not value:
+        warnings.warn("{attr} is ignored")
```
Comment (webknjaz) [inline setuptools/dist.py:294]: I think this should be
```suggestion
        warnings.warn("{attr} is ignored", DistDeprecationWarning)
```
(following the style in this module).

File: changelog.d/2777.misc.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Build does not fail fast when ``use_2to3`` is supplied but set to a false value.
```
Comment (webknjaz) [inline changelog.d/2777.misc.rst:1]: I think, this should be a bugfix, not a misc change.

**Review by webknjaz — COMMENTED** [review]
In general, I think it's a move in the right direction but could you also add a regression test for this scenario? I'd like to see both positive and negative test cases added.

File: changelog.d/2777.misc.rst (hunk 2 above)
Comment (jaraco) [reply changelog.d/2777.misc.rst:1]: Bugfix is misc. Naming is weird.

File: setup.cfg [hunk 3]
```diff
@@ -124,7 +124,7 @@ distutils.setup_keywords =
 	dependency_links = setuptools.dist:assert_string_list
 	test_loader = setuptools.dist:check_importable
 	test_runner = setuptools.dist:check_importable
-	use_2to3 = setuptools.dist:invalid
+	use_2to3 = setuptools.dist:invalid_ignored_if_false
```
Comment (jaraco) [inline setup.cfg:127]:
```suggestion
	use_2to3 = setuptools.dist:invalid_if_true
```

File: changelog.d/2777.misc.rst (hunk 2 above)
Comment (webknjaz) [reply changelog.d/2777.misc.rst:1]: Ah, I see. The towncrier config in this repo doesn't declare a separate bugfix type. Many other projects do so. This is what confused me :)

**Comment by webknjaz** [comment]
FTR somebody needs to evaluate this as a possible fix for #2776.

**Comment by jaraco** [comment]
> should 58.0.2 be yanked? I assume this will go into a 58.0.3

What value would there be in yanking 58.0.2?

**Comment by jaraco** [comment]
> In general, I think it's a move in the right direction but could you also add a regression test for this scenario? I'd like to see both positive and negative test cases added.

I'm happy to skip regression tests for these tweaks.

**Comment by haoyuanli** [comment]
> > should 58.0.2 be yanked? I assume this will go into a 58.0.3
> 
> What value would there be in yanking 58.0.2?

No, there wouldn't be, if a 58.0.3 is released.

**Comment by potiuk** [comment]
We have some packages as dependencies, which have 2_to_3 set to true (specifically FlaskOpenId  which has not been updated since 2016, so this is likely it won't be updated. https://github.com/mitsuhiko/flask-openid/ Any chance setuptools can simply ignore 2_to_3 setting, rather than ignore when false ?

**Comment by plumdog** [comment]
@potiuk See https://github.com/pypa/setuptools/issues/2775#issuecomment-914606508 and https://github.com/pypa/setuptools/issues/2769 for rationale behind not ignoring the flag - the installed package almost certainly won't be importable.

**Comment by potiuk** [comment]
Yeah. Absolutely. Actually we had an interesting case because the Flask-openid actually was working with just removing 2_to_3 - because it was really small, the changes were harmless (like removing u` from unicod strings) and the only non-compatible change was under "if python < 2" anyway. Thankfully we got response from the maintainers and we became maintainers ourselves and we released a new version of package (after manually converting it with 2_to_3 and making it python3 - only). We also released a .wh;l package which remove the need of using setuptools at all.

I agree it's not possible and tha approach you come up with with "false" is good.
